src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,646 @@ const ActorInit En_Rz_InitVars = {
+        }
+
+        this->animIndex = animIndex;
+    }
+}
+
+void func_80BFB9E4(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void func_80BFBA1C(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, 0.0f);
+}
+
+s32 func_80BFBA50(EnRz* this, PlayState* play) {
+    if (EN_RZ_GET_PATH(&this->actor) != 0x3F) {
+        this->path = &play->setupPathList[EN_RZ_GET_PATH(&this->actor)];
+        if (this->path != NULL) {
+            Path* path = this->path;
+            Vec3s* points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+            f32 temp_fv0 = points->x - this->actor.world.pos.x;
+            f32 temp_fv1 = points->z - this->actor.world.pos.z;
+
+            if ((SQ(temp_fv0) + SQ(temp_fv1)) < SQ(10.0f)) {
+                this->unk_3BC = 0;
+                this->unk_420 &= ~1;
+            } else {
+                this->unk_3BC = path->count - 1;
+                this->unk_420 |= 1;
+                return true;
+            }
+        }
+    } else {
+        this->path = NULL;
+        this->unk_3BC = 0;
+    }
+    return false;
+}
+
+s32 func_80BFBB44(EnRz* this) {
+    Path* path = this->path;
+    Vec3s* pointPos;
+    f32 temp_fa0;
+    f32 temp_fa1;
+
+    if (path == NULL) {
+        return true;
+    }
+
+    pointPos = &((Vec3s*)Lib_SegmentedToVirtual(path->points))[this->unk_3BC];
+    temp_fa0 = pointPos->x - this->actor.world.pos.x;
+    temp_fa1 = pointPos->z - this->actor.world.pos.z;
+    this->actor.world.rot.y = Math_Atan2S(temp_fa0, temp_fa1);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 2000, 200);
+
+    if ((SQ(temp_fa0) + SQ(temp_fa1)) < SQ(10.0f)) {
+        if (this->unk_420 & 1) {
+            this->unk_3BC--;
+            if (this->unk_3BC < 0) {
+                return 2;
+            }
+        } else {
+            this->unk_3BC++;
+            if (this->unk_3BC >= path->count) {
+                return 2;
+            }
+        }
+        return 1;
+    }
+    return 0;
+}
+
+s32 EnRz_CanTalk(EnRz* this, PlayState* play) {
+    if ((this->actor.xzDistToPlayer < 80.0f) && (fabsf(this->actor.playerHeightRel) < 20.0f) &&
+        Player_IsFacingActor(&this->actor, 0x3000, play)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_80BFBCEC(EnRz* this, PlayState* play) {
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+
+    if ((this->actor.xzDistToPlayer < 100.0f) && Actor_IsFacingPlayer(&this->actor, 0x3000) &&
+        player->stateFlags2 & PLAYER_STATE2_2000000) {
+        return true;
+    }
+
+    return false;
+}
+
+void EnRz_UpdateSkelAnime(EnRz* this, PlayState* play) {
+    if (this->animIndex < 7) {
+        SkelAnime_Update(&this->skelAnime);
+    } else {
+        LinkAnimation_Update(play, &this->skelAnime);
+    }
+}
+
+Actor* func_80BFBDA0(EnRz* this, PlayState* play) {
```

Comment:
```suggestion
EnRz* func_80BFBDA0(EnRz* this, PlayState* play) {
```
To go along with the struct member change, this functino only ever returns an `EnRz*`, so I think we can be more specific. Would need to cast `npc` when it is returned though.